modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/repositories/NotebooksRepository.java

```diff
@@ -21,58 +21,71 @@
 import java.util.TreeSet;
 import java.util.UUID;
 import org.apache.ignite.Ignite;
+import org.apache.ignite.console.db.ForeignKeyIndex;
 import org.apache.ignite.console.db.OneToManyIndex;
 import org.apache.ignite.console.db.Table;
 import org.apache.ignite.console.dto.Notebook;
 import org.apache.ignite.console.tx.TransactionManager;
 import org.apache.ignite.transactions.Transaction;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
 /**
  * Repository to work with notebooks.
  */
 @Repository
-public class NotebooksRepository extends AbstractRepository<Notebook> {
+public class NotebooksRepository {
+    /** */
+    private final TransactionManager txMgr;
+
     /** */
     private final Table<Notebook> notebooksTbl;
 
     /** */
     private final OneToManyIndex notebooksIdx;
 
+    /** */
+    private final ForeignKeyIndex accFkIdx;
+
     /**
      * @param ignite Ignite.
      * @param txMgr Transactions manager.
      */
-    @Autowired
     public NotebooksRepository(Ignite ignite, TransactionManager txMgr) {
-        super(ignite, txMgr);
+        this.txMgr = txMgr;
 
         notebooksTbl = new Table<Notebook>(ignite, "wc_notebooks")
             .addUniqueIndex(Notebook::getName, (notebook) -> "Notebook '" + notebook.getName() + "' already exits");
 
         notebooksIdx = new OneToManyIndex(ignite, "wc_account_notebooks_idx");
+
+        accFkIdx = new ForeignKeyIndex(ignite, "wc_accounts_fk_idx");
```

Comment:
What is the difference between the ForeignKeyIndex and the OneToManyIndex? It looks the same

Reply:
No, they are different.  OneToManyIndex holds <UUID to set of UUIDs>. And ForeignKeyIndex holds <UUID to UUID> links. Also ForeignKeyIndex  validate add/get/remove operations.